test/unit/app/services/asset_manager/attachment_updater_test.rb

```diff
@@ -4,281 +4,234 @@ class AssetManager::AttachmentUpdaterTest < ActiveSupport::TestCase
-                        .with(thumbnail_asset.asset_manager_id, attachment_data, nil, { "draft" => false })
+        expected_attribute_hash = {
+          "access_limited_organisation_ids" => edition.organisations.map(&:content_id),
+        }
 
-          updater.call(attachment_data, draft_status: true)
+        assets.each do |asset|
+          AssetManager::AssetUpdater.expects(:call).with(asset.asset_manager_id, attachment.attachment_data, nil, expected_attribute_hash)
         end
+
+        AssetManager::AttachmentUpdater.call(attachment.attachment_data, access_limited: true)
       end
     end
 
-    describe "link header updates" do
-      let(:edition) { FactoryBot.create(:published_edition) }
-      let(:parent_document_url) { edition.public_url }
-      let(:attachment) { FactoryBot.create(:file_attachment, file: simple_pdf, attachable: edition) }
+    context "when attachment's attachable is published" do
+      let(:edition) { create(:published_news_article) }
+      let(:attachment) { create(:file_attachment, attachable: edition) }
+      let(:assets) do
+        [
+          Asset.new(asset_manager_id: "original_asset_manager_id", variant: Asset.variants[:original], filename: "simple.pdf"),
+          Asset.new(asset_manager_id: "thumbnail_asset_manager_id", variant: Asset.variants[:thumbnail], filename: "thumbnail_simple.pdf.png"),
+        ]
+      end
 
-      context "when attachment doesn't belong to an edition" do
-        let(:attachment) { FactoryBot.create(:file_attachment) }
+      before { attachment.attachment_data.assets = assets }
 
-        it "does not update status of any assets" do
-          update_service.expects(:call).never
+      it "marks corresponding assets as not draft" do
+        expected_attribute_hash = {
+          "draft" => false,
+        }
 
-          updater.call(attachment_data, link_header: true)
+        assets.each do |asset|
+          AssetManager::AssetUpdater.expects(:call).with(asset.asset_manager_id, attachment.attachment_data, nil, expected_attribute_hash)
         end
-      end
 
-      context "when attachment belongs to a draft edition" do
-        let(:draft_edition) { FactoryBot.create(:draft_edition) }
-        let(:attachment) { FactoryBot.create(:file_attachment, file: simple_pdf, attachable: draft_edition) }
-        let(:parent_document_url) { draft_edition.public_url(draft: true) }
+        AssetManager::AttachmentUpdater.call(attachment.attachment_data, draft_status: true)
+      end
 
-        it "sets parent_document_url for attachment using draft hostname" do
-          update_service.expects(:call)
-                          .with(original_asset.asset_manager_id, attachment_data, nil, { "parent_document_url" => parent_document_url })
-          update_service.expects(:call)
-                        .with(thumbnail_asset.asset_manager_id, attachment_data, nil, { "parent_document_url" => parent_document_url })
+      it "sets parent_document_url for all assets" do
```

Comment:
See my comment on line 41 about these test names matching to convey similar behaviour